fuel-core/src/schema/tx/types.rs

```diff
@@ -214,8 +216,20 @@ impl Transaction {
 
     async fn status(&self, ctx: &Context<'_>) -> async_graphql::Result<Option<TransactionStatus>> {
         let db = ctx.data_unchecked::<Database>();
-        let status = db.get_tx_status(&self.0.id())?;
-        Ok(status.map(Into::into))
+
+        let txpool = ctx.data::<Arc<TxPool>>().unwrap().pool();
+
+        let transaction_in_pool = txpool.find(&[self.0.id()]).await;
```

Comment:
You can just extend TxPool to support `find_one` function it should be easy to add as you just need to see how `find` is made, this should simplify a bit logic here.
Other than that this seems okay.